File: 프로그래머스/lv3/49191. 순위/순위.kt
class Solution {
    fun solution(n: Int, results: Array<IntArray>): Int {
        val arr = MutableList(n) { IntArray(n) }
        
        results.forEach {
            arr[it[0] - 1][it[1] - 1] = 1
            arr[it[1] - 1][it[0] - 1] = -1
        }
        
        for (k in 0 until n) {
            for (i in 0 until n) {
                for (j in 0 until n) {
                    if (i != j && arr[i][j] == 0 && arr[i][k] == 1 && arr[k][j] == 1) {
                        arr[i][j] = 1
                        arr[j][i] = -1
                        arr[j][k] = -1
                        arr[k][i] = -1
                    }
                }
            }
        }
        
        return arr.count { it.count { it == 0 } == 1 }
    }
}

File: 프로그래머스/lv3/49191. 순위/순위.py
def solution(n, results):
    answer = 0
    arr = [[0] * n for _ in range(n)]
    
    for a, b in results:
        arr[a - 1][b - 1] = 1
        arr[b - 1][a - 1] = -1
        
    for k in range(n):
        for i in range(n):
            for j in range(n):
                if i == j or arr[i][j] in [1, -1]:
                    continue
                
                if arr[i][k] == arr[k][j] == 1:
                    arr[i][j] = 1
                    arr[j][i] = arr[k][i] = arr[j][k] = -1
                    
    for i in arr:
        if i.count(0) == 1:
            answer += 1
    
    return answer